ctsit/nacculator
# Add support for reading from multiple REDCap projects

**What does this change do?** 

Adds a [combine_adrc_projects.py](./nacc/combine_adrc_projects.py) script.

**Why was this change made?** 

To support reading from multiple projects since the NACC forms have been broken out into multiple REDCap projects.

## Verification

1. Import four [projects xml](https://drive.google.com/drive/folders/1z5bQMbvym7_rHhiBKg1pepo5gzz1tV4w) into your local REDcap instance
2. Create api tokens for each project
3. Create a copy of the config file: `cp nacculator_cfg.ini.example example.config.ini`
4. Update `redcap_server` with your local redcap server (`http://localhost:<port>/api/`)
5. Add the four api tokens to tokens (do not include spaces after the comma) to `tokens` in `example.config.ini`
6. Enable the virtual environment: `source .venv/bin/activate`
7.  Install dependencies: `pip install .`
8. Run the script: `python nacc/combine_adrc_projects.py example.config.ini`

Verify that a `redcap_input.csv` file was generated in the root project directory that contains the records for all projects.

## Affirmations

All of these should have a check by them. Any exception requires an explanation.
